lib/compress/zstd_compress.c

```diff
@@ -6296,6 +6333,29 @@ ZSTD_copySequencesToSeqStoreExplicitBlockDelim(ZSTD_CCtx* cctx,
         ZSTD_storeSeq(&cctx->seqStore, litLength, ip, iend, offBase, matchLength);
         ip += matchLength + litLength;
     }
+
+    /* If we skipped repcode search while parsing, we need to update repcodes now */
+    assert(idx >= startIdx);
+    if (externalRepSearch == ZSTD_ps_disable && idx != startIdx) {
```

Comment:
I think you meant `if (externalRepSearch == ZSTD_ps_enable &&`

Reply:
No, I changed it after renaming the cParam. Disable = don't search for external repcodes.

Reply:
I believe this test implies that, when reaching that `if ()` statement, `externalRepSearch != ZSTD_ps_auto`.
And I'm sure it is, but this condition is ensured elsewhere, so this is a long distance correlation.
This is probably worth an `assert()`.

Reply:
Will add the assert before merging!